File: android/app/src/main/java/com/liive/ride/designsystem/LiiveGlassPanel.kt
//  LiiveGlassPanel.kt  ·  Liive Ride DS (Compose)  ·  mirrors components/ride/GlassPanel
//  Frosted material panel over the live map for HUD chips and voice controls.
package com.liive.ride.designsystem

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.unit.Dp

enum class GlassMaterial { Thin, Regular, Thick }

@Composable
fun LiiveGlassPanel(
    modifier: Modifier = Modifier,
    material: GlassMaterial = GlassMaterial.Regular,
    shape: RoundedCornerShape = LiiveRadius.lg,
    padding: Dp = LiiveSpacing.m + LiiveSpacing.xs2,
    content: @Composable () -> Unit
) {
    val c = LiiveTheme.colors
    val materialColor = when (material) {
        GlassMaterial.Thin -> c.materialThin
        GlassMaterial.Regular -> c.materialRegular
        GlassMaterial.Thick -> c.materialThick
    }

    Box(
        modifier
            .shadow(LiiveElevation.hud, shape, clip = false)
            .clip(shape)
            .background(materialColor)
            .border(LiiveSpacing.xs2 / 4, c.borderStrong, shape)
            .padding(padding)
    ) { content() }
}

File: Packages/RideSharingFeature/Sources/RideSharingFeature/DesignSystem/LiiveGlassPanel.swift
//  LiiveGlassPanel.swift  ·  Liive Ride DS (SwiftUI)
//  Frosted-glass panel that floats over the map (.ultraThin/.thin/.thick).

import SwiftUI

public struct LiiveGlassPanel<Content: View>: View {
    public enum Material { case thin, regular, thick }
    var material: Material = .regular
    var cornerRadius: CGFloat = LiiveRadius.lg
    var padding: CGFloat = LiiveSpacing.m + LiiveSpacing.xs2
    let content: Content

    public init(material: Material = .regular, cornerRadius: CGFloat = LiiveRadius.lg,
                padding: CGFloat = LiiveSpacing.m + LiiveSpacing.xs2, @ViewBuilder content: () -> Content) {
        self.material = material; self.cornerRadius = cornerRadius
        self.padding = padding; self.content = content()
    }

    private var blur: SwiftUI.Material {
        switch material {
        case .thin: return .ultraThinMaterial
        case .regular: return .thinMaterial
        case .thick: return .regularMaterial
        }
    }

    private var materialFill: Color {
        switch material {
        case .thin: return LiiveColor.materialThin
        case .regular: return LiiveColor.materialRegular
        case .thick: return LiiveColor.materialThick
        }
    }

    public var body: some View {
        let shape = RoundedRectangle(cornerRadius: cornerRadius, style: .continuous)

        content
            .padding(padding)
            .background(materialFill, in: shape)
            .background(blur, in: shape)
            .overlay(
                shape.strokeBorder(LiiveColor.borderStrong, lineWidth: LiiveSpacing.xs2 / 4)
            )
            .liiveShadow(.hud)
    }
}
